### PR TITLE
Add testing module

### PR DESCRIPTION
**Pull request by ry**
I think once this is landed we can change the main repo to use this.

### CODE REVIEW

**Comment by bartlomieju** [comment]
Offtopic: is `mod.ts` a preferred name for main file in Deno module?

**Comment by ry** [comment]
@bartlomieju yes - index.ts comes with the wrong connotations - and main.ts should be reserved for executable programs. The “mod.ts” follows Rust’s pattern, is shorter, and doesn’t come with any preconceived notions about how it might work. I’m hoping we can start a style guide soon that will describe this and other conventions.

**Comment by bartlomieju** [comment]
@ry thanks for explanation, that makes sense. Style guide is very much desired as Deno is getting more recognition.

File: testing/mod.ts [hunk 1]
```diff
@@ -0,0 +1,175 @@
+/*!
+   Copyright 2018 Propel http://propel.site/.  All rights reserved.
```
Comment (bartlomieju) [inline testing/mod.ts]: Isn't this copyright contradictory to MIT in root directory?

Comment (ry) [reply testing/mod.ts]: Good catch. Fixed.

**Pull request merged**